module2/homework/transformContainers/transform.cpp

```diff
@@ -0,0 +1,28 @@
+#include <algorithm>
+#include <iostream>
+
+#include "tranform.hpp"
+
+std::map<int, std::string> removeDuplicateAndTranformToMap(
+    std::list<std::string> uniqueList,
+    std::deque<int> uniqueDeque) {
+    uniqueList.sort();
+    uniqueList.erase(std::unique(uniqueList.begin(), uniqueList.end()), uniqueList.end());
+
+    for (const auto el : uniqueDeque) {
+        std::cout << el << ' ';
+    }
+
+    std::sort(uniqueDeque.begin(), uniqueDeque.end());
+    uniqueDeque.erase(std::unique(uniqueDeque.begin(), uniqueDeque.end()), uniqueDeque.end());
+
+    std::map<int, std::string> result{};
+    std::transform(uniqueDeque.begin(),
+                   uniqueDeque.end(),
+                   uniqueList.begin(),
+                   std::inserter(result, result.end()),
```

Comment:
you can use:
`std::back_inserter(result)`

Reply:
it doesnt compile

Reply:
you're right. back_inserter doesn't work for string, sorry